IPython/lib/display.py

```diff
@@ -383,7 +383,7 @@ def __init__(self,
         result_html_suffix : str
             text to append at the end of link [default: '<br>']
         """
-        if isdir(path):
+        if Path(path).is_dir():
             raise ValueError("Cannot display a directory using FileLink. "
               "Use FileLinks to display '%s'." % path)
         self.path = fsdecode(path)
```

Comment:
Would it work if we initialize `self.path` as a `Path` object?

Reply:
I will try to make it work if i can.